orchard/modules/_branch.py

```diff
@@ -6,6 +6,228 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+import pprint
+import yaml
+from pathlib import Path
+import os
 
-class Branch:
-    pass
+
+# Gets a named module from a link or config file
+def get_module(file_data, name):
+    return next((item for item in file_data['modules'] if item['name'] == name), None)
+
+
+# Returns the depth level of a given module based on the link file
+def depth_level(link_file_data, module):
+    current_depth = 0
+    if 'dependencies' in module:
+        for mod_name in module['dependencies']:
+            current_depth = max(current_depth, depth_level(link_file_data, get_module(link_file_data, mod_name)))
+    return current_depth + 1
+
+
+# Returns true if an argument causes branching in a link module
+def is_branching_arg(link_file_mod, argname):
+    for arg in link_file_mod['arguments']:
+        if arg['name'] == argname:
+            if 'isBranch' in arg and not arg['isBranch']:
+                return False
+            else:
+                return True
+    return True
+
+
+# Goes through a list and returns true if the given argument branches, recursively checking in 'exclusive' blocks
+def is_branching_in_list(arglist, name):
+    for arg in arglist:
+        if 'exclusive' in arg:
+            if not is_branching_in_list(arg['exclusive'], name):
+                return False
+        if 'name' not in arg:
+            continue
+        if arg['name'] == name:
+            if 'isBranch' in arg and not arg['isBranch']:
+                return False
+            else:
+                return True
+    return True
+
+
+def check_args(module_1, module_2, link_module):
+    # For each argument in the first module
+    if 'arguments' in module_1:
+        if 'arguments' not in module_2:
+            return False
+        for arg in module_1['arguments']:
+            argname = list(arg.keys())[0]
```

Comment:
Each argument dictionary contains, and will only ever contain a single entry. As such I'm always guaranteed to get the key I want, because that's the only key that exists in the given dictionary (and if the config file is changed from that then all of this parsing is going to break anyways).

If you've got a better way to efficiently get the sole key from a dictionary, find the dictionary with a sole key that matches from a list of dict's, and compare the values for each key for equality then I'm all ears.